### PR TITLE
improved sample conf to cover all options present in the manpage

### PR DESCRIPTION
**Pull request by verfriemelt-dot-org**
i tried to include all the examples given in the mainpage and sorted them a bit around.
im not sure on all the default values given, maybe you could double check.

### CODE REVIEW

**Comment by yshui** [comment]
This is great work, thanks a lot!

A few problems:
1) I want to keep the examples in the sample config. e.g.
```
blur-background-exclude = [
	"window_type = 'dock'",
	"window_type = 'desktop'",
	"_GTK_FRAME_EXTENTS@:c"
];
```
They could be included as part of the comments.

2) The `wintypes` option should be included. It's a config file only option, that's why you probably have missed it.

Can you make sure options in the old sample.conf are all still included in the new one?

**Comment by verfriemelt-dot-org** [comment]
> They could be included as part of the comments.

done

> Can you make sure options in the old sample.conf are all still included in the new one?

you mean as the default settings, when the sample config is used?

**Comment by verfriemelt-dot-org** [comment]
the reasoning behind the things follows these rules:

1) group settings with a section started with a block like this:

```
#################################
#             Shadows           #
#################################
```
2) between options there is one blank line

3) if the option description is online one line, the next line is the option with its default value

```
# The top offset for shadows, in pixels. (defaults to -15)
# shadow-offset-y = -15
```

4) if the description is more than one line, the option is seperated by one empty comment line

```
# Don't draw shadows on drag-and-drop windows. This option is deprecated, 
# you should use the *wintypes* option in your config file instead.
#
# no-dnd-shadow = false
```

5) examples within a description are indented by two spaces:

```
# Specify a list of conditions of windows that should have no shadow.
# multiple exceptions: 
#   shadow-exclude = [
#     "name = 'Notification'",
#     "class_g = 'Conky'",
#     "class_g ?= 'Notify-osd'",
#     "class_g = 'Cairo-clock'",
#     "_GTK_FRAME_EXTENTS@:c"
#   ];
#
# single exception:
#   shadow-exclude = "n:e:Notification";
#
# shadow-exclude = []
```

File: picom.sample.conf [hunk 1]
```diff
@@ -1,81 +1,406 @@
...
+#   shadow-exclude 'bounding_shaped'
+# or 
+#   shadow-exclude 'bounding_shaped && !rounded_corners'
+# instead.
+#
+# shadow-ignore-shaped = ''
+
+# Specify a list of conditions of windows that should have no shadow.
+# multiple exceptions: 
+#   shadow-exclude = [
```
Comment (yshui) [inline picom.sample.conf]: This example is duplicated below.

File: picom.sample.conf [hunk 2]
```diff
@@ -1,81 +1,406 @@
...
-# shadow-exclude-reg = "x10+0+0";
-# xinerama-shadow-crop = true;
 
-# Opacity
+# Specify a X geometry that describes the region in which shadow should not
+# be painted in, such as a dock window region. Use 
+#    shadow-exclude-reg = "x10+0+0"
+# for example, if the 10 pixels on the bottom of the screen should not have shadows painted on.
+#
+# shadow-exclude-reg =
```
Comment (yshui) [inline picom.sample.conf]:
```suggestion
# shadow-exclude-reg = ""
```

File: picom.sample.conf [hunk 3]
```diff
@@ -1,81 +1,406 @@
...
+# shadow-green = 0
+
+# Blue color value of shadow (0.0 - 1.0, defaults to 0).
+# shadow-blue = 0
+
+# Do not paint shadows on shaped windows. Note shaped windows 
+# here means windows setting its shape through X Shape extension. 
+# Those using ARGB background is beyond our control. 
+# Deprecated, use 
+#   shadow-exclude 'bounding_shaped'
```
Comment (yshui) [inline picom.sample.conf]: Missing `=` here.

File: picom.sample.conf [hunk 4]
```diff
@@ -1,81 +1,406 @@
...
+# blur-kern = ''
 blur-kern = "3x3box";
-# blur-kern = "5,5,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1";
-# blur-background-fixed = true;
+
+
+# Exclude conditions for background blur.
+# example:
+#
+#    blur-background-exclude = [
```
Comment (yshui) [inline picom.sample.conf]: This example is duplicated below.

File: picom.sample.conf [hunk 5]
```diff
@@ -1,81 +1,406 @@
...
+# log-file = '/path/to/your/log/file'
+
+# Show all X errors (for debugging)
+# show-all-xerrors = false
+
+# Write process ID to a file.
+# write-pid-path = '/path/to/your/log/file'
 
 # Window type settings
+#
```
Comment (yshui) [inline picom.sample.conf]: Can you copy the man page section about `wintypes` here?

**Comment by verfriemelt-dot-org** [comment]
i missed somehow the section about the window type settings :)

**Comment by yshui** [comment]
Looks good!

**Comment by verfriemelt-dot-org** [comment]
thank you! :)

**Pull request merged**